components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -206,23 +206,34 @@ func (r *NotebookReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	} else {
 		// Got the pod
 		podFound = true
-		if len(pod.Status.ContainerStatuses) > 0 &&
-			pod.Status.ContainerStatuses[0].State != instance.Status.ContainerState {
-			log.Info("Updating container state: ", "namespace", instance.Namespace, "name", instance.Name)
-			cs := pod.Status.ContainerStatuses[0].State
-			instance.Status.ContainerState = cs
-			oldConditions := instance.Status.Conditions
-			newCondition := getNextCondition(cs)
-			// Append new condition
-			if len(oldConditions) == 0 || oldConditions[0].Type != newCondition.Type ||
-				oldConditions[0].Reason != newCondition.Reason ||
-				oldConditions[0].Message != newCondition.Message {
-				log.Info("Appending to conditions: ", "namespace", instance.Namespace, "name", instance.Name, "type", newCondition.Type, "reason", newCondition.Reason, "message", newCondition.Message)
-				instance.Status.Conditions = append([]v1beta1.NotebookCondition{newCondition}, oldConditions...)
+
+		if len(pod.Status.ContainerStatuses) > 0 {
+			notebookContainerFound := false
+			for i := range pod.Status.ContainerStatuses {
+				if pod.Status.ContainerStatuses[i].Name == instance.Name &&
+					pod.Status.ContainerStatuses[i].State != instance.Status.ContainerState {
+					log.Info("Updating Notebook CR state: ", "namespace", instance.Namespace, "name", instance.Name)
+					cs := pod.Status.ContainerStatuses[i].State
+					instance.Status.ContainerState = cs
+					oldConditions := instance.Status.Conditions
+					newCondition := getNextCondition(cs)
+					// Append new condition
+					if len(oldConditions) == 0 || oldConditions[0].Type != newCondition.Type ||
+						oldConditions[0].Reason != newCondition.Reason ||
+						oldConditions[0].Message != newCondition.Message {
+						log.Info("Appending to conditions: ", "namespace", instance.Namespace, "name", instance.Name, "type", newCondition.Type, "reason", newCondition.Reason, "message", newCondition.Message)
+						instance.Status.Conditions = append([]v1beta1.NotebookCondition{newCondition}, oldConditions...)
+					}
+					err = r.Status().Update(ctx, instance)
+					if err != nil {
+						return ctrl.Result{}, err
+					}
+					notebookContainerFound = true
+					break
+				}
 			}
-			err = r.Status().Update(ctx, instance)
-			if err != nil {
-				return ctrl.Result{}, err
+			if !notebookContainerFound {
+				log.Info("WARNING: Notebook container is not found, so could not update State of Notebook CR")
```

Comment:
I don't think that a log.Info should emit a `WARNING` message.
If the message is indeed a warning, use `log.Warn`.
This sounds like an error though.

Reply:
I agree with @yanniszark.
Although AFAIK `logr` does not have a [Warning level](https://github.com/go-logr/logr#why-not-more-named-levels-like-warning). So we should do this an Error instead.

@gilbeckers could you also make the error message more explicit as to why it couldn't find the Notebook Container? Something like: `Could not find the Notebook container. No container has the same name with the CR '{container-name}'. Will not update the status of the CR.`